src/PowerShellEditorServices/Server/PsesDebugServer.cs

```diff
@@ -95,6 +97,12 @@ public void Dispose()
         {
             _powerShellContextService.IsDebugServerActive = false;
             _jsonRpcServer.Dispose();
+            _serverStart.SetResult(true);
+        }
+
+        public async Task WaitForShutdown()
+        {
+            await _serverStart.Task;
```

Comment:
If this is server start, why does it get set on shutdown? Should it be something like `serverStopped` instead?

Reply:
Yeah I guess it should be called like `_serverRunning` or something.

Reply:
changed it to that because when the "result is set" it's like saying set the result of the "running". Let me know if you feel stronger about `_serverStopped`

Reply:
Changed to `_serverStopped`!